### PR TITLE
feat/add-action-for-docs-deploy

### PR DESCRIPTION
**Pull request by huhn511**
# Description of change

Please write a summary of your changes and why you made them.

## Links to any relevant issues

Be sure to reference any related issues by adding `fixes issue #`.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/deploy_docs_to_gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup mdBook
+        uses: peaceiris/actions-mdbook@v1
+        with:
+          mdbook-version: 'latest'
+
+      - run: mdbook build
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
```
Comment (nothingismagick) [inline .github/workflows/deploy_docs_to_gh-pages.yml]: Any third party actions that use our credentials need to be audited and potentially forked.

**Review by nothingismagick — APPROVED** [review]

File: .github/workflows/deploy_docs_to_gh-pages.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_branch: gh-pages
+          publish_dir: ./book
```
Comment (nothingismagick) [inline .github/workflows/deploy_docs_to_gh-pages.yml]: Should it be `./book` or `./docs`?

Comment (huhn511) [reply .github/workflows/deploy_docs_to_gh-pages.yml]: ./book

The ./docs dir has the markdown files and these gets bundled together to html files. So website is in ./book

File: .github/workflows/deploy_docs_to_gh-pages.yml (hunk 1 above)
Comment (huhn511) [reply .github/workflows/deploy_docs_to_gh-pages.yml]: Fixed with last commit https://github.com/iotaledger/stronghold.rs/pull/132/commits/937287ce731a1483557211525998ccde1cf55c79

**Pull request merged**